File: mqtt/src/subscribe.js
const mqtt = require("mqtt");

const brokerUrl = "mqtt://127.0.0.1:2222";
const client = mqtt.connect(brokerUrl);

client.on("connect", () => {
  console.log("Connected to MQTT broker");
  client.subscribe("sensor/temperature", (err) => {
    if (!err) {
      console.log("Subscribed to topic: sensor/temperature");
    }
  });
});

client.on("message", (topic, message) => {
  console.log(`Received message: ${message.toString()} on topic: ${topic}`);
});

File: mqtt/src/examples/subscribe.py
import paho.mqtt.client as mqtt


def on_connect(client, userdata, flags, rc):
    if rc == 0:
        print("Connected to the broker.")
    else:
        print("Failed to connect, return code: " + str(rc))

    client.subscribe("data")


def on_message(client, userdata, msg):
    print(f"\n{msg.topic}: {msg.payload.decode()}")


client = mqtt.Client()

client.on_connect = on_connect
client.on_message = on_message

client.connect("localhost", 2222, 60)

client.loop_start()

try:
    while True:
        pass

except KeyboardInterrupt:
    print("\nQuitting...")

finally:
    client.loop_stop()
    client.disconnect()
